### PR TITLE
Space - Antonia

### PR DESCRIPTION
**Pull request by antoniairizarry**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | It provides data to the database.
Describe in your own words what the Controller is doing in Rails | The controller works with requests and data to determine which action will be taken, which will be given back to Views.
Describe in your own words what the View is doing in Rails | View renders data from controller to be user-facing.
Describe an edge-case controller test you wrote | N/A
What is the purpose of using strong params? (i.e. the params method in the controller) | It makes it safer to work with data that comes from forms and can make code more succinct.
How are Rails migrations related to Rails models? | Changes in the Rails model require migrations because they cause changes in the schema.
Describe one area of Rails that are still unclear on | Strong params, and still getting used to all the details you have to keep track of while using Rails.

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  def create
+    @task = Task.new(
+      name: params[:task][:name],
+      description: params[:task][:description],
+      created_at: params[:task][:created_at])
+      if @task.save
+        redirect_to task_path(@task.id)
+        return
+      else
+        render :new, :bad_request
```
Comment (CheezItMan) [inline app/controllers/tasks_controller.rb:28]: I think this works, but I like to do this being more explicit about the status code.

```suggestion
        render :new, status: :bad_request
```

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  <head>
+    <title>TaskList</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:12]: I suggest adding navigation links like making a new task and linking to the all tasks page.

File: app/views/tasks/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<h3>New Task<h3>
+<%= form_with model: @task, class: 'create-task' do |f| %>
+<p>Enter your new task:</p>
+<%= f.label :name %>
+<%= f.text_field :name%>
+<%= f.label :description %>
+<%= f.text_area :description %>
+<%= f.submit "Save task", class: "button" %>
```
Comment (CheezItMan) [inline app/views/tasks/new.html.erb:8]: I realize this came after tasklist, just a reminder in future projects you can dry up the forms with a partial view.

**Comment by CheezItMan** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | You've created the views, they're not really styled
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ⚠ , no mark tasking complete

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements |






## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Perfect Indentation | ✅
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅



## Summary


Nicely done, you hit the main learning goals here.  Take a look at my few comments and let me know what questions you have.
